### ISSUE TITLE
Feature request: Userspace reboot logo

### ISSUE THREAD

**Issue by ghh-jb** [opa334/Dopamine#308]
Add a custom userspace reboot image(like in coolstar jailbreaks)

**Comment by 4193883-eng** [comment]
i think it isn't possible without a bootrom exploit

**Comment by opa334** [comment]
We planned on doing this, but our implementation does not work on A14+ currently so we delayed it to a later update

**Comment by eglacias** [comment]
Yeah I’ve always wanted just to see the equivalent of the checkmate image instead of the Lucifer fall from Paradise Apple bite when my phone reboot :-)

**Comment by ssdsl0126** [comment]
> We planned on doing this, but our implementation does not work on A14+ currently so we delayed it to a later update

When do you plan to launch?

**Comment by ahiezx** [comment]
Checkra1n did it before i don't doubt dopamine doing it

**Comment by ghh-jb** [comment]
@opa334 what about this feature? Any plans?

**Comment by opa334** [comment]
will be added soon! :)

**Comment by eglacias** [comment]
> will be added soon! :)

Oh man, that’s great! I’d love to see the check mate crossed with dopamine icon, I can’t stand seeing the bite out of the apple logo as often as possible I would love to see that gone :-) seems trivial, but it’s aesthetically very, very pleasing :-)

**Comment by ghh-jb** [comment]
> will be added soon! :) 

Very cool, thank you!

**Comment by hostedbyjustus** [comment]
Opa out here continuing perfecting an already perfect Jailbreak to the point where it's becoming THE Jailbreak out there

**Comment by hostedbyjustus** [comment]
@opa334 Hope you enjoy your time without Jailbreak development but is there an update you can give on this feature? Would also be interesting to hear what your next step with Dopamine is. 

Though if the project ever comes to an end I propose you do not announce that in a big way 😅

**Comment by eglacias** [comment]
dunno why I'm so stoked about it, but inspriation comes on every reboot (: 
i wish i could replace the respring spinner too, i HATE spinner hypnotizers...
the moon shot inspired a whole generation... inspiration is everything!
tho i never EVER see it reboot or respring with this stable jailbreak, never crashes
then again i do lose battery sometimes and...
feel like I've overcome satan when i see the checkmate logo instead if ever...

**Comment by opa334** [comment]
I have recently pushed the feature to the `bootlogo` branch. Figuring out how to deal with iPads (that can boot in landscape) was a massive PITA that took me months to solve, but in the latest commit it should work. The problem is I don't have any test devices and testing in general has been quite limited so far.

**Comment by hostedbyjustus** [comment]
> I have recently pushed the feature to the `bootlogo` branch. Figuring out how to deal with iPads (that can boot in landscape) was a massive PITA that took me months to solve, but in the latest commit it should work. The problem is I don't have any test devices and testing in general has been quite limited so far.

I unfortunately only really have two devices that would have the ability to test this on (12 mini on 15.5b4 and X on 16.3.1) but if it's helpful, I could help beta testing. Though I guess you're looking for test iPads instead.

**Comment by eglacias** [comment]
i got a 8plus on 16.61, but ill have to figure out how to compile the boot logo source!

**Comment by opa334** [comment]
You don't need to compile, the pipeline already did: https://github.com/opa334/Dopamine/actions/runs/20572617328

**Comment by hostedbyjustus** [comment]
My testing concluded that the feature works so far. That did include testing every color combination. 

Though I did notice a difference when running on my 12 mini on 15.5b4 in comparision to my X on 16.3.1 which would be that the 12 mini only shows the custom bootlogo for a bit before changing into the Apple logo while the X shows the custom bootlogo for the entire duration of the boot. I suppose that was the only way you could get it running on A14+ considering you mentioned issues with these devices. 

I guess the remaining problems are caused by iPads now. 

One question I did have was, whether this feature will be optional considering I already know that not everyone wants features that are forced onto the user. Otherwise it's a great feature which would almost make a 2.5 version name viable.

**Comment by eglacias** [comment]
I would just put it into the regular release personally, who wouldn’t want the awesomeness of clearly showing what it was that they just triggered. And to overcoming the fall from paradise…

Then again, personally, I would fancy having an option to use the check rain logo :-) checkmate Against the evil system

**Comment by hostedbyjustus** [comment]
> I would just put it into the regular release personally, who wouldn’t want the awesomeness of clearly showing what it was that they just triggered. And to overcoming the fall from paradise…
> 
> Then again, personally, I would fancy having an option to use the check rain logo :-) checkmate Against the evil system

What?

**Comment by eglacias** [comment]
ahh nevermind i see custom boot logo (wowon't ad a png from photos likely needs a specifit size and format (jpeg?
16.6.1 shows logo, blinks shows logo again, then white apple for last 60 sec of boot
....btw i have NEVER spent time on a tirvial feature before always worrying about missing nuts and bolts features...
i also was deluded in that of course this only shows on jailbreak not first boot...
i swear jailbreak used to fail having to retry 30% of the time, this one succeeds every time

**Comment by ghh-jb** [comment]
Nice! I am going to test it right now.

**Comment by ghh-jb** [comment]
Worked perfectly on:
iPhone 11, iOS 16.3.1

Thank you for this amazing feature!

**Comment by eglacias** [comment]
Only works in the first half of the boot on iOS 16.6.1 on an 8+ white apple second half

**Comment by ghh-jb** [comment]
Have you ever used Taurine jailbreak? It is done like there. That is normal I think.

**Comment by opa334** [comment]
The logo is drawn once and persists until SpringBoard takes over the framebuffer

**Comment by ghh-jb** [comment]
> The logo is drawn once and persists until SpringBoard takes over the framebuffer

Is it launchd reexec when the logo blinks once?

**Comment by eglacias** [comment]
> Have you ever used Taurine jailbreak? It is done like there. That is normal I think.

ahh yes i have a taurine device... yep it blinks the logo 2x then... white apple fair enough!
maan i want the neptune and monarch theme for dopamine tho...
then again, i only have to EVER jailbreak if i run out of battery, can go for unprecedented months probably years without crashing!

**Comment by opa334** [comment]
> > The logo is drawn once and persists until SpringBoard takes over the framebuffer
> 
> Is it launchd reexec when the logo blinks once?

yes
